src/ec/packages/ec-component-blockquote/blockquote.story.js

```diff
@@ -8,18 +9,48 @@ import data from '@ecl/ec-specs-blockquote/demo/data';
 import blockquote from './ecl-blockquote.html.twig';
 import notes from './README.md';
 
+// Labels for the group ids
+// const optionalGroupId = 'Optional elements';
+const requiredGroupId = 'Mandatory elements';
+const useCasesGroup = 'Use cases';
+
+// Preserve the adapted specs.
+const defaultData = { ...data };
+
+// declare your toggled data
+const authorBtnToggler = () => {};
+
+const prepareQuote = prepareData => {
+  // declare your knobs when you format your data
+  const citation = prepareData.citation
+    ? text('Citation', prepareData.citation, requiredGroupId)
+    : false;
+
+  const author = prepareData.author
+    ? text('Author', prepareData.author, requiredGroupId)
+    : false;
+
+  const newQuote = {
```

Comment:
I do see this approach as a bit redundant, why do you have to build to separated constants, redefining entirely the object to pass to the template?
The idea is that you pass the data to this function and it should return this data formatted, in this case where don't have any button, it is straightforward.
You basically receive the data and inside of it you are only going to "override" what you need and return it. I'm attaching here how i would make the story file, up to you to use it as such or make it your own.. ;)
```suggestion
/* eslint-disable no-param-reassign, no-shadow */
import { storiesOf } from '@storybook/html';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withNotes } from '@ecl-twig/storybook-addon-notes';
import withCode from '@ecl-twig/storybook-addon-code';

import data from '@ecl/ec-specs-blockquote/demo/data';

import blockquote from './ecl-blockquote.html.twig';
import notes from './README.md';

// Labels for the group ids
const optionalGroupId = 'Optional elements';
const requiredGroupId = 'Mandatory elements';

const prepareQuote = data => {
  data.citation = text('citation', data.citation, requiredGroupId);
  data.author = text('author', data.author, requiredGroupId);

  return data;
};

storiesOf('Components/Blockquote', module)
  .addDecorator(withKnobs)
  .addDecorator(withCode)
  .addDecorator(withNotes)
  .add(
    'default',
    () => {
      const storyData = prepareQuote(data);
      storyData.demo = text(
        'optional elements',
        'No optional element in this story',
        optionalGroupId
      );

      return blockquote(prepareQuote(storyData));
    },
    {
      notes: { markdown: notes, json: data },
    }
  );
```

Reply:
@Joosthe , i would really like to close this, i gave you this suggestion but you did it a bit differently, ok for me, but:
now you have the "demo" parameter in the notes tab, which should not be there and in my suggestion would not have appeared. 
This is  because you are still "cloning" data into another variable, (with a typo: "preperaData") and the demo property you set on the storyData var get assigned also to the main data object.